run/serializers.py

```diff
@@ -0,0 +1,69 @@
+from rest_framework import serializers
+from rest_framework.parsers import JSONParser
+from .models import Run
+from .models import FilePath
+from storage.models import File
+from storage.serializers import FileSerializer
+
+
+class FilePathSerializer(serializers.ModelSerializer):
+    file = FileSerializer(many=False)
+
+    # definition = FileDefinitionSerializer()
+
+    class Meta:
+        model = FilePath
+        fields = ('file',
+                  # 'definition',
+                  )
+
+
+class RunReportSerializer(serializers.ModelSerializer):
+    file_path_set = FilePathSerializer(many=True, read_only=True)
+
+    class Meta:
+        model = Run
+        fields = (
+            # 'game',
+            'id', 'status', 'end_time', 'log', 'file_path_set')
+        read_only_fields = fields
+
+
+class FilePathSetSerializerField(serializers.Field):
+    def to_representation(self, value):
+        raise NotImplemented()
+
+    def to_internal_value(self, data):
+        internal_value = []
+        if not isinstance(data, dict):
+            raise serializers.ValidationError('files data must be a dictionary.')
+        for file_definition in data:
+            if 'id' not in data[file_definition]:
```

Comment:
Here FilePaths are not really saved in the database, the save operation is done under RunCreateSerializer's control. I used it just like a datastucture just instead of tuple for example.